src/stream/src/executor/source/source_executor.rs

```diff
@@ -736,6 +828,29 @@ impl<S: StateStore> SourceExecutor<S> {
     }
 }
 
+async fn build_source_stream_and_poll_barrier(
```

Comment:
nit: maybe this should be called
```suggestion
async fn build_source_stream_or_receive_barrier(
```